### PR TITLE
feat: reorganize ISS wiring

### LINKED ISSUE #11683: Massage ISS detection/handling wiring
- incremental step of #10899 

- There are currently many calls into the `IssDetector`, from different locations
- The end goal is to only require a single call into the `IssDetector`, for any given `state+round` pair
- The changes to include in this ticket include:
    - moving the ISS wiring into the `PlatformWiring` class, to match other system wiring
    - collect all calls into the `IssDetector` in a single location, so that these calls can easily be collapsed into a single call 
    - move some miscellaneous ISS handling logic out of lambdas, and into the `IssHandler`

### PR DESCRIPTION
**Pull request by alittley**
- closes #11683 

[wiring diagram](https://mermaid.ink/svg/JSV7aW5pdDogeydmbG93Y2hhcnQnOiB7J2RlZmF1bHRSZW5kZXJlcic6ICdlbGsnfX19JSUKZmxvd2NoYXJ0IFRECnYwWy8iQ29uc2Vuc3VzIEV2ZW50IFN0cmVhbSIvXQpzdWJncmFwaCB2MVsiQ29uc2Vuc3VzIFBpcGVsaW5lIl0KdjJbIkNvbnNlbnN1cyBFbmdpbmUiXQp2M1siaW5PcmRlckxpbmtlcjxiciAvPvCfjIAiXQp2NCgoIvCfjIAiKSkKdjUoKCLwn5OsIikpCnY2KCgi8J+avSIpKQplbmQKc3ViZ3JhcGggdjdbIkV2ZW50IENyZWF0aW9uIl0KdjhbImV2ZW50Q3JlYXRpb25NYW5hZ2VyPGJyIC8+4p2k77iP8J+MgCJdCnY5WyJmdXR1cmVFdmVudEJ1ZmZlcjxiciAvPvCfjIAiXQp2MTB7eyJmdXR1cmVFdmVudEJ1ZmZlclNwbGl0dGVyIn19CmVuZApzdWJncmFwaCB2MTFbIkV2ZW50IEhhc2hpbmciXQp2MTJbWyJldmVudEhhc2hlciJdXQp2MTNbInBvc3RIYXNoQ29sbGVjdG9yIl0KZW5kCnN1YmdyYXBoIHYxNFsiRXZlbnQgVmFsaWRhdGlvbiJdCnYxNVsiZXZlbnREZWR1cGxpY2F0b3I8YnIgLz7wn4yAIl0KdjE2WyJldmVudFNpZ25hdHVyZVZhbGlkYXRvcjxiciAvPvCfjIAiXQp2MTdbImludGVybmFsRXZlbnRWYWxpZGF0b3IiXQplbmQKc3ViZ3JhcGggdjE4WyJHb3NzaXAiXQp2MTl7eyJnb3NzaXAifX0KdjIwWyJzaGFkb3dncmFwaDxiciAvPvCfjIDwn5OsIl0KZW5kCnN1YmdyYXBoIHYyMVsiSGVhcnRiZWF0Il0KdjIyWyJoZWFydGJlYXQiXQp2MjMoKCLinaTvuI8iKSkKZW5kCnYyNFsiT3JwaGFuIEJ1ZmZlcjxiciAvPvCfjIAiXQpzdWJncmFwaCB2MjVbIlBDRVMgUmVwbGF5Il0KdjI2Wy8icGNlc1JlcGxheWVyIi9dCnYyNygoIuKchSIpKQplbmQKc3ViZ3JhcGggdjI4WyJQcmVjb25zZW5zdXMgRXZlbnQgU3RyZWFtIl0KdjI5Wy8iZXZlbnREdXJhYmlsaXR5TmV4dXMiL10KdjMwWy8icGNlc1NlcXVlbmNlciIvXQp2MzFbInBjZXNXcml0ZXI8YnIgLz7inIXwn4yA8J+TgPCfmr0iXQp2MzIoKCLwn5WRIikpCmVuZApzdWJncmFwaCB2MzNbIlNpZ25hdHVyZSBNYW5hZ2VtZW50Il0Kc3ViZ3JhcGggdjM0WyJJc3MgRGV0ZWN0b3IiXQp2MzV7eyJleHRyYWN0U2lnbmF0dXJlc0Zvcklzc0RldGVjdG9yIn19CnYzNlsiaXNzRGV0ZWN0b3IiXQp2MzdbLyJpc3NIYW5kbGVyIi9dCnYzOFsvImlzc05vdGlmaWNhdGlvbkVuZ2luZSIvXQp2Mzl7eyJpc3NOb3RpZmljYXRpb25TcGxpdHRlciJ9fQp2NDBbLyJzdGF0dXNNYW5hZ2VyX3N1Ym1pdENhdGFzdHJvcGhpY0ZhaWx1cmUiL10KZW5kCnY0MVsiU3RhdGUgU2lnbmF0dXJlIENvbGxlY3Rpb24iXQp2NDJbInN0YXRlU2lnbmVyIl0KZW5kCnY0M1siU3RhdGUgRmlsZSBNYW5hZ2VtZW50Il0Kc3ViZ3JhcGggdjQ0WyJTdGF0ZSBNb2RpZmljYXRpb24iXQp2NDVbImNvbnNlbnN1c1JvdW5kSGFuZGxlcjxiciAvPvCflK7wn5WRIl0KdjQ2e3sicnVubmluZ0hhc2hVcGRhdGUifX0KZW5kCnN1YmdyYXBoIHY0N1siVHJhbnNhY3Rpb24gUHJlaGFuZGxpbmciXQp2NDhbWyJhcHBsaWNhdGlvblRyYW5zYWN0aW9uUHJlaGFuZGxlciJdXQp2NDkoKCLwn5SuIikpCmVuZAp2NTBbImhhc2hMb2dnZXIiXQp2NTFbLyJsYXRlc3RDb21wbGV0ZVN0YXRlTmV4dXMiL10KdjUyWyJsYXRlc3RDb21wbGV0ZVN0YXRlU2NoZWR1bGVyIl0KdjUzWy8idHJhbnNhY3Rpb25Qb29sIi9dCnY1NCgoIvCfk4AiKSkKdjIgLS0gInJvdW5kcyIgLS0+IHYwCnYyIC0tICJyb3VuZHMiIC0tPiB2NDUKdjIgLS4gIm5vbi1hbmNpZW50IGV2ZW50IHdpbmRvdyIgLi0+IHY0CnYyIC0tICJmbHVzaCByZXF1ZXN0IiAtLT4gdjYKdjI0IC0tICJwcmVjb25zZW5zdXMgZXZlbnRzIiAtLT4gdjQxCnYyNCAtLSAicHJlY29uc2Vuc3VzIGV2ZW50cyIgLS0+IHY0OAp2MjQgLS0gInByZWNvbnNlbnN1cyBldmVudHMiIC0tPiB2OQp2MjQgLS0gImV2ZW50cyB0byBzZXF1ZW5jZSIgLS0+IHYzMAp2NDMgLS0gIm1pbmltdW0gaWRlbnRpZmllciB0byBzdG9yZSIgLS0+IHY1NAp2NDEgLS0gInN0YXRlcyIgLS0+IHY0Mwp2NDEgLS0gImNvbXBsZXRlIHN0YXRlIiAtLT4gdjUxCnY0MSAtLSAiY29tcGxldGVkIHJlc2VydmVkIHNpZ25lZCBzdGF0ZSB0byBub3RpZnkiIC0tPiB2NTIKdjQ4IC0uICJmdXR1cmVzIiAuLW8gdjQ5CnY4IC0uICJub24tdmFsaWRhdGVkIGV2ZW50cyIgLi0+IHYxNwp2MTUgLS0gImV2ZW50cyB3aXRoIHVudmFsaWRhdGVkIHNpZ25hdHVyZXMiIC0tPiB2MTYKdjI5IC0uICJ3YWl0IGZvciBkdXJhYmlsaXR5IiAuLW8gdjMyCnYxMiAtLSAiaGFzaGVkIGV2ZW50cyIgLS0+IHYxMwp2MTYgLS0gInVub3JkZXJlZCBldmVudHMiIC0tPiB2MjQKdjM1IC0tICJwb3N0IGNvbnNlbnN1cyBzaWduYXR1cmVzIiAtLT4gdjM2CnY5IC0tICJwb3NzaWJsZSBwYXJlbnQgbGlzdHMiIC0tPiB2MTAKdjEwIC0tICJwb3NzaWJsZSBwYXJlbnRzIiAtLT4gdjgKdjE5IC0tICJldmVudHMgdG8gaGFzaCIgLS0+IHYxMgp2MjIgLS0gImhlYXJ0YmVhdCIgLS0+IHYyMwp2MyAtLSAibGlua2VkIGV2ZW50cyIgLS0+IHYyCnYzIC0tICJldmVudHMgdG8gZ29zc2lwIiAtLT4gdjUKdjE3IC0tICJub24tZGVkdXBsaWNhdGVkIGV2ZW50cyIgLS0+IHYxNQp2MzYgLS0gImlzcyBub3RpZmljYXRpb25zIiAtLT4gdjM5CnYzOSAtLSAiaXNzIG5vdGlmaWNhdGlvbiIgLS0+IHYzNwp2MzkgLS0gIklTUyBub3RpZmljYXRpb24iIC0tPiB2MzgKdjM5IC0tICJJU1Mgbm90aWZpY2F0aW9uIiAtLT4gdjQwCnYyNiAtLSAiZXZlbnRzIHRvIGhhc2giIC0tPiB2MTIKdjI2IC0tICJkb25lIHN0cmVhbWluZyBwY2VzIiAtLT4gdjI3CnYzMCAtLSAidW5saW5rZWQgZXZlbnRzIiAtLT4gdjMKdjMwIC0tICJldmVudHMgdG8gd3JpdGUiIC0tPiB2MzEKdjMxIC0tICJsYXRlc3QgZHVyYWJsZSBzZXF1ZW5jZSBudW1iZXIiIC0tPiB2MjkKdjEzIC0tICJub24tdmFsaWRhdGVkIGV2ZW50cyIgLS0+IHYxNwp2NDYgLS0gInJ1bm5pbmcgaGFzaCB1cGRhdGUiIC0tPiB2MAp2NDYgLS0gInJ1bm5pbmcgaGFzaCB1cGRhdGUiIC0tPiB2NDUKdjQyIC0tICJzdGF0ZSBzaWduYXR1cmUgdHJhbnNhY3Rpb24iIC0tPiB2NTMKY2xhc3NEZWYgczAgZmlsbDojY2NjLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjAsdjEwLHYxOSx2MjYsdjI5LHYzMCx2MzUsdjM3LHYzOCx2MzksdjQwLHY0Nix2NTEsdjUzIHMwCmNsYXNzRGVmIHMxIGZpbGw6IzlDRixzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxLHYxMSx2MTQsdjE4LHYyMSx2MjUsdjI4LHYzMyx2NDQsdjQ3LHY3IHMxCmNsYXNzRGVmIHMyIGZpbGw6I2ZmOSxzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxMix2MTMsdjE1LHYxNix2MTcsdjIsdjIwLHYyMix2MjQsdjMsdjMxLHYzNix2NDEsdjQyLHY0Myx2NDUsdjQ4LHY1MCx2NTIsdjgsdjkgczIKY2xhc3NEZWYgczMgZmlsbDojZjg4LHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjIzLHYyNyx2MzIsdjQsdjQ5LHY1LHY1NCx2NiBzMwpjbGFzc0RlZiBzNCBmaWxsOiNCRUYsc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MzQgczQK?bgColor=e8e8e8)

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -621,18 +627,13 @@ public class SwirldsPlatform implements Platform {
...
-        final OutputWire<IssNotification> issOutput =
-                platformWiring.getIssDetectorWiring().issNotificationOutput();
-        issOutput.solderTo(
-                "issNotificationEngine", "ISS notification", n -> notificationEngine.dispatch(IssListener.class, n));
-        issOutput.solderTo("statusManager_submitCatastrophicFailure", "ISS notification", n -> {
-            if (Set.of(IssType.SELF_ISS, IssType.CATASTROPHIC_ISS).contains(n.getIssType())) {
-                platformStatusManager.submitStatusAction(new CatastrophicFailureAction());
-            }
-        });
-        issOutput.solderTo("issHandler", "ISS notification", issHandler::issObserved);
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:635]: why move these into the ISS handler?

Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:635]: @lpetrovic05 These actions are part of how we handle an ISS: I think it's a lot more understandable and maintainable to have the responses to an ISS collected in a single spot. I don't think we gain anything from indirectly calling these methods via the wiring framework.

Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:635]: @lpetrovic05 Bullet points 1 and 2 have been completed, so this is ready for a re-review

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulers.java [hunk 2]
```diff
@@ -258,6 +260,10 @@ public static PlatformSchedulers create(
                         .withMetricsBuilder(model.metricsBuilder().withUnhandledTaskMetricEnabled(true))
                         .build()
                         .cast(),
+                model.schedulerBuilder("issHandler")
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulers.java:263]: do we need a separate scheduler for the iss handler?

**Review by lpetrovic05 — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**